### PR TITLE
Fetch data for board cards at build time server-side

### PR DESCRIPTION
**Pull request by farisashai**
# Description

No visible changes to website - board card data comes from spreadsheet instead of hardcoded JSON now so its always updated and can easily be changed

Set ISR to regenerate data on website once every day (arbitrary choice)

Specific spreadsheet URL is an environment variable in deployment settings (find it on Vercel or ask me if you need to run locally) 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Logistical Change (something about code process / a readme change / documentation update)
- [ ] CI Change (relating to deployment / continuous integration)
- [ ] Something Else 

# How Has This Been Tested?
- [x] Tested on Localhost
- [x] Tested on Deployed Preview

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code as needed, particularly in hard-to-understand areas
- [x] My changes generate no new warnings (if they do, please explain why)

### CODE REVIEW

File: pages/about.tsx [hunk 1]
```diff
@@ -29,9 +30,69 @@ const AboutPage: React.FC = () => {
...
+          profile_image: getImage(userData),
+          email: getUCSDEmail(userData),
+          linkedin_link: getLinkedIn(userData),
+          personal_link: getWebsite(userData),
+        }))
+        .filter((user) => user.name !== "") // Filter out missing rows
+        .filter((user) => user.org !== ""), // Filter out members at large
+    },
+  };
+}
```
Comment (steets250) [inline pages/about.tsx:98]: Should this be a util function instead of in the `about.tsx` page?

File: pages/about.tsx [hunk 2]
```diff
@@ -29,9 +30,69 @@ const AboutPage: React.FC = () => {
...
+        .map((row) => row.c) // Map row object to row data array
+        .map((userData) => ({
+          name: getName(userData),
+          org: getOrg(userData),
+          title: getTitle(userData),
+          specialization: getSpecialization(userData),
+          profile_image: getImage(userData),
+          email: getUCSDEmail(userData),
+          linkedin_link: getLinkedIn(userData),
+          personal_link: getWebsite(userData),
```
Comment (farisashai) [inline pages/about.tsx:92]: created these helper functions mostly just to keep the return value object readable and use if statements but we can just inline it too lmk

File: pages/about.tsx (hunk 1 above)
Comment (farisashai) [reply pages/about.tsx]: good idea just moved it

File: src/api/BoardAPI.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  return get(userData[0]).toLowerCase();
+};
+
+const getTitle = (userData) => {
+  if (!get(userData[1]) || get(userData[0] === "-----")) return "";
+  return get(userData[1]);
+};
+
+const getSpecialization = (userData) => {
+  if (!get(userData[3]) || get(userData[3]) === "-----") return "";
```
Comment (trulyronak) [inline src/api/BoardAPI.ts:16]: confused by these numbers / where they come from?

File: src/api/BoardAPI.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+};
+
+const getWebsite = (userData) => get(userData[14]);
+
+export const getBoardData = async () => {
+  const url = process.env.SPREADSHEET_URL;
+
+  const rawData: { c: { v: any }[] }[] = await fetch(url) // This is the type thats given cant change
+    .then((res) => res.text())
+    .then((res) => JSON.parse(res.substring(47).slice(0, -2))?.table?.rows); // strip weird data header
```
Comment (trulyronak) [inline src/api/BoardAPI.ts:39]: why 47? can you include in a comment an example of the line somewhere so it's clear where these numbers are coming from

File: pages/about.tsx [hunk 5]
```diff
@@ -29,9 +31,19 @@ const AboutPage: React.FC = () => {
...
 
 export default AboutPage;
+
+export async function getStaticProps() {
+  const boardData = await getBoardData();
+  return {
+    props: {
+      board: boardData || [],
+    },
+    revalidate: 1 * 60 * 60 * 24, // once every day (in seconds)
```
Comment (trulyronak) [inline pages/about.tsx:47]: question - is there a way to manually trigger a revalidate / if we add something to the board spreadsheet, is there a way to trigger an instantaneous change?

File: src/api/BoardAPI.ts (hunk 3 above)
Comment (trulyronak) [reply src/api/BoardAPI.ts:16]: for this section of code in general, i think to allow for more code maintainbility, it'd be good to explain the general process for these (all the `gets` here seem to have the same pattern), and to potentially include in that general comment an example of a line. Alot of these numbers seem completely arbitrary and this section in general seems too complicated to maintain without the needed context

**Review by trulyronak — CHANGES_REQUESTED** [review]
left some comments regarding code maintainability, logic seems good tho

**Comment by farisashai** [comment]
board data population is now resilient to upper/lower case headers, moving, and adding columns on the spreadsheet. renaming a column header completely or deleting it would still break some functionality

**Review by trulyronak — CHANGES_REQUESTED** [review]
almost there, just have a few requests

**Review by trulyronak — APPROVED** [review]
poggers

**Pull request merged**